src/lib/plugin_apis/lvm.api

```diff
@@ -1242,6 +1343,20 @@ BDLVMLVdata* bd_lvm_lvinfo (const gchar *vg_name, const gchar *lv_name, GError *
  */
 BDLVMLVdata** bd_lvm_lvs (const gchar *vg_name, GError **error);
 
+/**
+ * bd_lvm_lvstree:
+ * @vg_name: (nullable): name of the VG to get information about LVs from
+ * @error: (out) (optional): place to store error (if any)
+ *
+ * This function will fill out the data_lvs, metadata_lvs, and segs fields as well.
```

Comment:
Yes, I am afraid that filling out the lists will change the space complexity of the result structure too much.  Existing callers would pay dearly for stuff that they don't need. I am hoping that between us we can figure out what is best.

Reply:
Ok, if it would be too slow to get these results, then it makes sense to not get it in `bd_lvm_lvinfo`, but I think it might be too confusing (at least it confused me) to have two different results for two different calls to get `BDLVMLVdata`, I would prefer a separate struct for these "extra" parameters (it's too many structs already, but I think it would be better that way). Thoughts @tbzatek?

Reply:
I don't think it would be too slow: We already iterate over all segments in both bd_lvm_lvinfo and bd_lvm_lvs (and use a O(n^2) linear search to find duplicates, urks). The "tree" variant gets a couple more fields from `lvs`, but that should be acceptable (unless you have a million stripes and the "devices" property etc gets really huge).

I am worried about the space needed to store the results, and I wanted to start carefully...

But I now think that we should have "tree" variants of both bd_lvm_lvs and bd_lvm_lvinfo, and if you think that the space overhead is acceptable also for existing callers that are not interested in the sub-structure, then I would be very happy to change the existing bd_lvm_lvinfo and bd_lvm_lvs functions to always fill segs, data_lvs, and metadata_lvs.

(Earlier I thought that a tree variant of bd_lvm_lvinfo is not very useful since you likely need information for all sub-lvs immediately, but if you are only interested in a single lv, it is better to make the necessary multiple calls to bd_lvm_lvinfo to retrieve the whole tree below a single LV than it is to retrieve info for absolutely all LVs, which might be many many more.)

Reply:
Ok, I guess we can do it this way. There's one other option -- we plan to do a new major release with API changes so we could change the API of `bd_lvm_lvs` and `bd_lvm_lvinfo`. But that's something we can decide when we're ready to release 3.0.